api/environments/models.py

```diff
@@ -118,8 +118,11 @@ def get_from_cache(cls, api_key):
                     "project__organisation",
                     "amplitude_config",
                 )
-                environment = cls.objects.select_related(*select_related_args).get(
-                    Q(api_key=api_key) | Q(api_keys__key=api_key)
+                environment = (
+                    cls.objects.select_related(*select_related_args)
+                    .filter(Q(api_key=api_key) | Q(api_keys__key=api_key))
```

Comment:
Yeah, I debated that for a little while. My thinking was that we still want to know if we're getting more than one here because it should never happen. If we use `.first()`, in a scenario where we get more than one different environment record we wouldn't ever know about it. So I'd rather throw an exception rather than hide it by using `.first()`. Does that make sense?